### ISSUE TITLE
SC1091: Use currently active path to check sourced files

### ISSUE THREAD

**Issue by ChillerDragon** [koalaman/shellcheck#2176]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I am not sure if this is in the scope of this project but it would be nice if shellcheck could search the files to be sourced not in the current directory but where they are actually loaded from. It would be nice if for example ``cd`` calls would be taken under consideration. My use case would be editing scripts from different path than they are run. Or shellcheck path independent sources with something like this:

```
#!/bin/bash

CURRENTPATH="$(pwd)"
SCRIPTPATH="$( cd -- "$(dirname "$0")" || exit 1 >/dev/null 2>&1 ; pwd -P )"
cd "$SCRIPTPATH" || exit 1

# shellcheck disable=SC1091
source lib/lib.sh

cd "$CURRENTPATH" || exit 1
```

#### Here's a snippet or screenshot that shows the problem:

```sh
[chiller@arch /tmp/shellcheck]$ tree .
.
├── lib
│   └── lib.sh
└── main.sh

1 directory, 2 files
[chiller@arch /tmp/shellcheck]$ cat main.sh 
#!/bin/bash

cd /tmp/shellcheck || exit 1
source lib/lib.sh

log "foo"

[chiller@arch /tmp/shellcheck]$ bash main.sh 
[*] foo
[chiller@arch /tmp/shellcheck]$ shellcheck -x main.sh 
[chiller@arch /tmp/shellcheck]$ cd lib/
[chiller@arch /tmp/shellcheck/lib]$ bash ../main.sh 
[*] foo
[chiller@arch /tmp/shellcheck/lib]$ shellcheck -x ../main.sh 

In ../main.sh line 4:
source lib/lib.sh
       ^--------^ SC1091: Not following: lib/lib.sh: openBinaryFile: does not exist (No such file or directory)

For more information:
  https://www.shellcheck.net/wiki/SC1091 -- Not following: lib/lib.sh: openBi...
```

#### Here's what shellcheck currently says:

In ../main.sh line 4:
source lib/lib.sh
       ^--------^ SC1091: Not following: lib/lib.sh: openBinaryFile: does not exist (No such file or directory)

#### Here's what I wanted or expected to see:

Nothing because shellcheck follows the ``cd`` and finds the file

**Comment by adoyle-h** [comment]
@koalaman 
This issue could be reproduced by below codes.

```sh
mkdir check
cat <<EOF > check/a.bash
#!/usr/bin/env bash

CUR_DIR=\$(pwd)
# shellcheck source=./b.bash
. "\$CUR_DIR/b.bash"
EOF
touch check/b.bash
```

```sh
shellcheck -x check/a.bash

In check/a.bash line 5:
. "$CUR_DIR/b.bash"
  ^---------------^ SC1091 (info): Not following: ./b.bash: openBinaryFile: does not exist (No such file or directory)

For more information:
  https://www.shellcheck.net/wiki/SC1091 -- Not following: ./b.bash: openBina...
```

```sh
cd check
shellcheck -x a.bash
# no error
```

-----
shellcheck version: 0.8.0
bash: GNU bash, version 5.2.2(1)-release (aarch64-apple-darwin21.6.0)
